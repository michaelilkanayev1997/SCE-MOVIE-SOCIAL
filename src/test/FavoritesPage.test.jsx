import React from "react";
import { shallow, configure } from "enzyme";
import FavoritesPage from "../pages/FavoritesPage ";
import Adapter from "@wojtekmaj/enzyme-adapter-react-17";
import "jest-extended";

configure({ adapter: new Adapter() });

describe("FavoritesPage", () => {
  it("should render the page without crashing", () => {
    const wrapper = shallow(<FavoritesPage />);
    expect(wrapper).toHaveLength(1);
    expect(wrapper).toMatchSnapshot();
  });

  it("should display loading spinner if user is not authenticated", () => {
    const wrapper = shallow(<FavoritesPage />);
    expect(wrapper.find("GlobalLoading")).toHaveLength(1);
  });

  it("should not display favorites if no user or there is loading ", () => {
    const wrapper = shallow(<FavoritesPage />);
    expect(wrapper.find(".card-container")).toHaveLength(0);
  });
});
