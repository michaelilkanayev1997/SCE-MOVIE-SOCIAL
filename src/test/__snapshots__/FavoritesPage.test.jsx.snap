// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FavoritesPage should render the page without crashing 1`] = `ShallowWrapper {}`;
